dao/src/main/java/org/wso2/testgrid/dao/repository/ProductRepository.java

```diff
@@ -147,4 +149,37 @@ public Product findByPrimaryKey(String id) throws TestGridDAOException {
         }
         return productTestStatuses;
     }
+
+    /**
+     * This method updates the last_success_timestamp column or last_failure_timestamp column with given timestamp.
+     * The column is selected by considering product status.
```

Comment:
Please provide a sample sql query you are using for reference purposes.. It's a bit hard to follow the string construction logic.

Also, please run _FindSecurityBugs_ for this, and make sure no issues pops up. You can do this with IntelliJ Idea itself.

Reply:
Fixed with https://github.com/wso2/testgrid/pull/803/commits/992f5f36ed390df948dff65148704178a4dbe307